src/text/raw_text_reader.rs

```diff
@@ -62,11 +63,12 @@ impl<T: TextIonDataSource> RawTextReader<T> {
         // If the reader's current value is the beginning of a container and the user calls `next()`,
         // we need to skip the entire container. We can do this by stepping into and then out of
         // that container; `step_out()` has logic that will exhaust the remaining values.
-        let need_to_skip_container = self
-            .current_value
-            .as_ref()
-            .map(|v| v.value().ion_type().is_container())
-            .unwrap_or(false);
+        let need_to_skip_container = !self.is_null()
```

Comment:
PR tour: this addressed a bug that caused the `RawTextReader` to try to step into `null` values with container types (`null.struct`, `null.list`, etc).